databases/week2/Part4.sql

```diff
@@ -0,0 +1,133 @@
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Employee` (
+  `id` INT NOT NULL,
+  `name` VARCHAR(45) NOT NULL,
+  `store_id` INT NOT NULL,
+  `phone` VARCHAR(45) NULL DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `store_id` (`store_id` ASC) VISIBLE,
+  CONSTRAINT `Employee_ibfk_1`
+    FOREIGN KEY (`store_id`)
+    REFERENCES `SupplyChain`.`Store` (`id`))
+ENGINE = InnoDB
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci;
+
+
+-- -----------------------------------------------------
+-- Table `SupplyChain`.`Supplier`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Supplier` (
+  `id` INT NOT NULL,
+  `name` VARCHAR(45) NULL DEFAULT NULL,
+  `phone` VARCHAR(45) NULL DEFAULT NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci;
+
+
+-- -----------------------------------------------------
+-- Table `SupplyChain`.`Products`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Products` (
+  `id` INT NOT NULL AUTO_INCREMENT,
+  `name` VARCHAR(45) NOT NULL,
+  `category_id` INT NOT NULL,
+  `supplier_id` INT NOT NULL,
+  `unit_price` INT NULL DEFAULT NULL,
+  `quantity_unit` INT NULL DEFAULT NULL,
+  `num_in_stock` INT NULL DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `supplier_id` (`supplier_id` ASC) VISIBLE,
+  CONSTRAINT `Products_ibfk_1`
+    FOREIGN KEY (`supplier_id`)
+    REFERENCES `SupplyChain`.`Supplier` (`id`))
+ENGINE = InnoDB
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci;
+
+
+-- -----------------------------------------------------
+-- Table `SupplyChain`.`Orders`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Orders` (
+  `id` INT NOT NULL,
+  `order_status` INT NULL DEFAULT NULL,
+  `due_order` DATETIME NULL DEFAULT NULL,
+  `made_by` INT NOT NULL,
+  `detail_id` INT NULL DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `detail_id` (`detail_id` ASC) VISIBLE,
+  INDEX `made_by` (`made_by` ASC) VISIBLE,
+  CONSTRAINT `Orders_ibfk_1`
+    FOREIGN KEY (`detail_id`)
+    REFERENCES `SupplyChain`.`Order_details` (`id`),
+  CONSTRAINT `Orders_ibfk_2`
+    FOREIGN KEY (`made_by`)
+    REFERENCES `SupplyChain`.`Store` (`id`))
+ENGINE = InnoDB
+DEFAULT CHARACTER SET = utf8mb4
+COLLATE = utf8mb4_0900_ai_ci;
+
+
+-- -----------------------------------------------------
+-- Table `SupplyChain`.`Order_details`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `SupplyChain`.`Order_details` (
+  `id` INT NOT NULL,
+  `total_items` INT NULL DEFAULT NULL,
+  `details` VARCHAR(45) NULL DEFAULT NULL,
+  `product_id` INT NOT NULL,
+  `order_id` INT NOT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `product_id` (`product_id` ASC) VISIBLE,
+  INDEX `order_id` (`order_id` ASC) VISIBLE,
+  CONSTRAINT `Order_details_ibfk_1`
```

Comment:
why do you need CONSTRAINT here?

Reply:
Another autocreation by workbench EER model

Reply:
@arya56 even if it is autocreated by the MySQL Workbench tool, it is important to know why those constraints are created, that way you get the best understanding of how MySQL works.